pkg/ocm/ocm_test.go

```diff
@@ -0,0 +1,147 @@
+package ocm_test
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"k8s.io/apimachinery/pkg/util/json"
+
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+	mocks "github.com/openshift/configuration-anomaly-detection/pkg/ocm/mock"
+)
+
+var _ = Describe("OCM", func() {
+	var (
+		mockCtrl          *gomock.Controller
+		client            *ocm.OcmClient
+		mocOCMConnection  *mocks.MockocmHandlerIf
+		clusterDeployment *hivev1.ClusterDeployment
+		awsAccountClaim   *awsv1alpha1.AccountClaim
+		err               error
+		clustername       string
+		supportRoleArn    string
+	)
+	Describe("When fetching a ClusterDeployment", func() {
+		BeforeEach(func() {
+			mockCtrl = gomock.NewController(GinkgoT())
+			mocOCMConnection = mocks.NewMockocmHandlerIf(mockCtrl)
+			client = &ocm.OcmClient{Comm: mocOCMConnection}
+			err = fmt.Errorf("some error")
+			clustername = "test-cluster"
+			clusterDeployment = &hivev1.ClusterDeployment{Spec: hivev1.ClusterDeploymentSpec{ClusterName: clustername}}
+		})
+		When("cluster ID is valid", func() {
+			It("should return a valid ClusterDeployment", func() {
+				cdstring, _ := json.Marshal(clusterDeployment)
+				mocOCMConnection.EXPECT().OcmGetResourceLive(clustername, "cluster_deployment").Return(string(cdstring), nil).Times(1)
```

Comment:
break this line down so it won't be too long?

Reply:
done